raimitigations/dataprocessing/sampler/synthesizer.py

```diff
@@ -535,27 +585,29 @@ def transform(
               the under_sampler parameter along with a float value for the ``strategy_under``
               parameter, an error will be raised;
               If None, the default value is set to "auto", which is the same as "minority".
+        :param label_col: the name or index of the label column;
         :return: the transformed dataset.
         :rtype: pd.DataFrame or np.ndarray
         """
-        self._check_if_fitted()
-        df, input_mode = self._arrange_transform_df(df, X, y)
-        if df is not None:
-            self._check_valid_df(df)
-            df = self._apply_transforms(df)
+        self._fit(X, y, df, label_col)
 
         if n_samples is None and conditions is not None:
             raise ValueError("ERROR: if 'conditions' is provided, the parameter 'n_samples' is also required.")
         if n_samples is not None:
             samples = self.sample(n_samples, conditions)
+        elif self.strategy is not None:
+            samples = self._generate_samples_strategy(self.df_info.df, self.strategy)
         else:
-            samples = self._generate_samples_strategy(df, strategy)
+            samples = self._generate_samples_strategy(self.df_info.df, strategy)
```

Comment:
It makes more sense to me to check first if the ``strategy`` parameter was provided. If it was, then use it instead of ``self.strategy``. If not, then use ``self.strategy``. Otherwise, if ``strategy`` is provided in the constructor, you won't be able to change the strategy afterwards.